snapcraft/project/_sanity_checks.py

```diff
@@ -32,7 +35,21 @@
     re.compile(r"^gui(/.*\.(png|svg|desktop))?$"),
 }
 
-# TODO: Add conduct_environment_sanity_check() to be run in the build environment
+
+def conduct_build_environment_sanity_check(provider: str):
+    if provider == "multipass":
+        _check_multipass_installed()
+
+
+def _check_multipass_installed():
+    if not shutil.which("multipass"):
+        if sys.platform == "linux":
+            if not shutil.which("snap"):
+                raise errors.SnapMissingLinuxError()
```

Comment:
nice covering this, we will always run from the snap moving forward, but this makes it easy to report on checks when running from source.